darts/models/theta.py

```diff
@@ -115,14 +120,241 @@ def predict(self, n: int) -> 'TimeSeries':
+            User-defined seasonality period. If not set, will be tentatively inferred from the training series upon
+            calling `fit()`.
+        model_mode
+            Type of model combining the Theta lines. Must be a Model Enum member. Defaults to `Model.ADDITIVE`.
+        season_mode
+            Type of seasonality. Must be a Season Enum member. Defaults to `Season.MULTIPLICATIVE`.
+        trend_mode
+            Type of trend to fit. Must be a Trend Enum member. Defaults to `Trend.LINEAR`.
+        normalization
+            If `True`, the data is normalized so that the mean is 1. Defaults to `True`.
+        """
+
+        super().__init__()
+
+        self.model = None
+        self.drift = None
+        self.mean = 1
+        self.length = 0
+        self.theta = theta
+        self.is_seasonal = False
+        self.seasonality = None
+        self.season_period = seasonality_period
+        self.model_mode = model_mode
+        self.season_mode = season_mode
+        self.trend_mode = trend_mode
+        self.wses = 0 if self.theta == 0 else (1 / theta)
+        self.wdrift = 1 - self.wses
+        self.fitted_values = None
+        self.normalization = normalization
+
+        raise_if_not(model_mode in Model,
+                     "Unknown value for model_mode: {}.".format(model_mode), logger)
+        raise_if_not(trend_mode in Trend,
+                     "Unknown value for trend_mode: {}.".format(trend_mode), logger)
+        raise_if_not(season_mode in Season,
+                     "Unknown value for season_mode: {}.".format(season_mode), logger)
+
+    def fit(self, ts, component_index: Optional[int] = None):
+        super().fit(ts, component_index)
+
+        self.length = len(ts)
+        # normalization of data
+        if self.normalization:
+            self.mean = ts.mean().mean()
+            raise_if_not(not np.isclose(self.mean, 0),
+                         "The mean value of the TimeSeries must not be 0 when using normalization", logger)
+            new_ts = ts / self.mean
+        else:
+            new_ts = ts.copy()
+
+        # Check for statistical significance of user-defined season period
+        # or infers season_period from the TimeSeries itself.
+        if self.season_mode is Season.NONE:
+            self.season_period = 1
+        if self.season_period is None:
+            max_lag = len(ts) // 2
+            self.is_seasonal, self.season_period = check_seasonality(ts, self.season_period, max_lag=max_lag)
+            logger.info('FourTheta model inferred seasonality of training series: {}'.format(self.season_period))
+        else:
+            # force the user-defined seasonality to be considered as a true seasonal period.
+            self.is_seasonal = self.season_period > 1
+
+        # Store and remove seasonality effect if there is any.
+        if self.is_seasonal:
+            _, self.seasonality = extract_trend_and_seasonality(new_ts, self.season_period,
```

Comment:
could you change `extract_trend_and_seasonality()` in order to accept Enums instead of strings? And also everywhere else it's needed.